rocks-lib/src/rockspec/build/builtin.rs

```diff
@@ -54,6 +61,98 @@ pub struct ModulePaths {
     pub libdirs: Vec<PathBuf>,
 }
 
+impl Build for BuiltinBuildSpec {
+    fn run(self, _rockspec: Rockspec, output_paths: TreeLayout, _no_install: bool) -> Result<()> {
+        // Detect all Lua modules
+        let modules = autodetect_modules()?
+            .into_iter()
+            .chain(self.modules)
+            .collect::<HashMap<_, _>>();
+
+        for (destination_path, module_type) in &modules {
+            match module_type {
+                ModuleType::SourcePath(source) => {
+                    let target = output_paths.src.join(PathBuf::from(
+                        destination_path.replace('.', std::path::MAIN_SEPARATOR_STR) + ".lua",
+                    ));
+
+                    std::fs::create_dir_all(target.parent().unwrap())?;
+
+                    std::fs::copy(source, target)?;
+                }
+                ModuleType::SourcePaths(files) => {
+                    let path = PathBuf::from(destination_path);
+                    std::fs::create_dir_all(path.parent().unwrap())?;
+
+                    cc::Build::new()
+                        .shared_flag(true)
+                        .files(files)
+                        .try_compile(destination_path)?;
+                }
+                ModuleType::ModulePaths(data) => {
+                    let path = PathBuf::from(destination_path);
+                    std::fs::create_dir_all(path.parent().unwrap())?;
+
+                    // TODO: Defines, libraries
+                    cc::Build::new()
+                        .shared_flag(true)
+                        .files(&data.sources)
+                        .includes(&data.incdirs)
```

Comment:
praise: woah, this looks so much cleaner than the lua implementation :star_struck:

Reply:
Had to add a bunch more flags for everything to work now but yeah it's very nice to work with :D